src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -180,7 +182,18 @@ Error DatasetManager::ProcessSetRequest(const Coap::Message &aMessage, SetReques
         // MGMT_ACTIVE_SET.req/MGMT_PENDING_SET.req from Commissioner
         // based on existing active dataset.
 
-        IgnoreError(Get<ActiveDatasetManager>().Read(aInfo.mDataset));
+        if (aCommand == kMgmtSet)
+        {
+            IgnoreError(Get<ActiveDatasetManager>().Read(aInfo.mDataset));
+        }
+    }
+
+    if (aCommand == kMgmtReplace)
+    {
+        // MGMT_ACTIVE_REPLACE can be used by commissioner.
+
+        VerifyOrExit(aInfo.mIsFromCommissioner);
+        VerifyOrExit(IsActiveDataset());
```

Comment:
Should we also validate that the Active Operational Dataset has the set of necessary TLVs?

Reply:
Good idea. Will add it in new push.

Reply:
Updated in new push. Added a new method `Dataset::ContainsAllRequiredTlvsFor()` to check if a dataset contains all the required TLVs for an Active or Pending Dataset. This method may prove useful in other places in the code.